cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -217,13 +217,33 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration) []string {
 			defaultArguments["audit-log-maxage"] = fmt.Sprintf("%d", *cfg.AuditPolicyConfiguration.LogMaxAge)
 		}
 	}
-
+	if cfg.APIServerExtraArgs == nil {
+		cfg.APIServerExtraArgs = map[string]string{}
+	}
+	cfg.APIServerExtraArgs["authorization-mode"] = getAuthzModes(cfg)
 	command = append(command, kubeadmutil.BuildArgumentListFromMap(defaultArguments, cfg.APIServerExtraArgs)...)
-	command = append(command, getAuthzParameters(cfg.AuthorizationModes)...)
 
 	return command
 }
 
+// getAuthzModes gets the authorization-related parameters to the api server
+// At this point, we can assume the list of authorization modes is valid (due to that it has been validated in the API
```

Comment:
I don't think we can assume it's valid actually. What's inside of `.*ExtraArgs` is by definition custom.

Reply:
I just kept the old comment. But actually its half true. While i tested, i realised, the configuration, coming from the file specified as --config=  is validated for RBAC, and Node existence. But its not true for extra-args, so i'll remove it.